logconsumer_test.go

```diff
@@ -181,3 +182,103 @@ func Test_ShouldRecognizeLogTypes(t *testing.T) {
 	c.Terminate(ctx)
 
 }
+
+func TestContainerLogWithErrClosed(t *testing.T) {
+	// First spin up a docker-in-docker container, then spin up an inner container within that dind container
+	// Logs are being read from the inner container via the dind container's tcp port, which can be briefly
+	// closed to test behaviour in connection-closed situations.
+	ctx := context.Background()
+
+	dind, err := GenericContainer(ctx, GenericContainerRequest{
+		Started: true,
+		ContainerRequest: ContainerRequest{
+			Image:        "docker:dind",
+			ExposedPorts: []string{"2375/tcp"},
+			Env:          map[string]string{"DOCKER_TLS_CERTDIR": ""},
+			WaitingFor:   wait.ForListeningPort("2375/tcp"),
+			Privileged:   true,
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer dind.Terminate(ctx)
+
+	remoteDocker, err := dind.Endpoint(ctx, "tcp")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := client.NewClientWithOpts(client.WithHost(remoteDocker))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	client.NegotiateAPIVersion(ctx)
+	provider := &DockerProvider{client: client}
+
+	nginx, err := provider.CreateContainer(ctx, ContainerRequest{Image: "nginx", ExposedPorts: []string{"80/tcp"}})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := nginx.Start(ctx); err != nil {
+		t.Fatal(err)
+	}
+	defer nginx.Terminate(ctx)
+
+	port, err := nginx.MappedPort(ctx, "80/tcp")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var consumer TestLogConsumer
+	if err = nginx.StartLogProducer(ctx); err != nil {
+		t.Fatal(err)
+	}
+	defer nginx.StopLogProducer()
+	nginx.FollowOutput(&consumer)
+
+	// Gather the initial container logs
+	time.Sleep(time.Second * 1)
```

Comment:
Do you foresee medium-term, potential issues with the sleep time not being enough?